src/content/blog/contributing-to-a-create-typescript-app-repository/index.mdx

```diff
@@ -0,0 +1,349 @@
+Try adding a few spaces after a `;` in a TypeScript source file such as `src/index.ts`, then saving.
+The spaces should be removed.
+
+You're now ready to start editing source files! 🎉
+
+## Building Changes
+
+The repository's `.github/DEVELOPMENT.md` should describe how to build source files into runnable output files.
+Commonly:
+
+```shell
+pnpm build --watch
+```
+
+I recommend also keeping the TypeScript compiler (`tsc`) running in a second, split terminal window as you make changes, so that you can get real-time updates from the type checker:
+
+```shell
+pnpm tsc --watch
+```
+
+Now, as you edit source files, equivalent output files should be made and continuously updated.
+
+## Validating Changes
+
+I'd recommend trying out your changes locally to make sure they're working as you'd expect.
+Two common situations for repositories are:
+
+- If the repository can be run in a terminal, try running your built files (most commonly, with a command like `node bin/index.js`)
+- If the repository is a dependency of another repository, try using [`pnpm link`](https://pnpm.io/cli/link) to try your changes locally
+  ```shell
+  # In the repository you're working on (say, create-typescript-app)
+  pnpm link --global
+  ```
+  ```shell
+  # In the repository consuming your repository
+  pnpm link create-typescript-app --global
+  ```
+
+The repository will also have automated tooling to validate changes.
+Those tools will also be describe in its `.github/DEVELOPMENT.md` file.
+
+Most repositories have the following commands:
+
+- `pnpm lint`: to run [ESLint](https://eslint.org)
+- `pnpm lint:knip`: to run [Knip](https://github.com/webpro/knip)
+- `pnpm lint:spelling`: to run [CSpell](https://cspell.org)
+
+### Running Tests
+
+Most repositories -mine included- generally ask that any changes to code logic should be accompanied by unit tests.
+Doing so makes sure that the logic is correct and stays correct as the repository changes over time.
+
+Tests can most commonly be run with something like:
+
+```shell
+pnpm test
+```
+
+#### Test Snapshots
+
+[Vitest snapshots](https://vitest.dev/guide/snapshot.html) are used in some repositories to help tests assert that some set of values is exactly what's expected.
+I generally use them when writing out the exact values would be annoying and cumbersome.
+You can generally update snapshots in repositories with a command like:
+
+```shell
+pnpm test -u
+```
+
+## Sending a Pull Request
+
+Once you've gotten as much confidence as you can in your changes, you'll want to send it in as a [Pull Request](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/about-pull-requests) for review.
+It's best to create a separate branch for your changes instead of sending the pull request from your `main` branch.
+
+I recommend giving the first commit on that branch a message adhering to [conventional commits].
```

Comment:
Is link to conventional commits missing?

Reply:
It is!